conans/client/graph/python_requires.py

```diff
@@ -21,6 +21,7 @@ def __init__(self, proxy, range_resolver):
         self._check_updates = False
         self._update = False
         self._remote_name = None
+        self.locked_versions = None
```

Comment:
Having a look at the way you are using this `locked_versions` attribute, I feel like it should be a `context_manager`:

```python
class ConanPythonRequire(object):
    ...
    @contextmanagere
    def lock_versions(self, versions_to_lock):
        old_cache = self._cached_requires
        try:
            self._cache_requires.update(  <data from versions_to_lock>)
            return
        catch:
        finally:
            self._cached_requires = old_cache
```

I think it will result in a cleaner implementation